src/Provider/Provider.Web/Orchestrators/AlertsOrchestrator.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System.Threading.Tasks;
+using Esfa.Recruit.Vacancies.Client.Application.Providers;
+using Esfa.Recruit.Vacancies.Client.Domain.Entities;
+using Esfa.Recruit.Vacancies.Client.Infrastructure.Client;
+
+namespace Esfa.Recruit.Provider.Web.Orchestrators
+{
+    public class AlertsOrchestrator
+    {
+        private readonly IRecruitVacancyClient _client;
+        private readonly ITimeProvider _timeProvider;
+
+        public AlertsOrchestrator(IRecruitVacancyClient client, ITimeProvider timeProvider)
+        {
+            _client = client;
+            _timeProvider = timeProvider;
+        }
+
+        public Task DismissAlert(VacancyUser user, AlertType alertType)
+        {
+            return _client.UpdateUserAlertAsync(user.UserId, alertType, _timeProvider.Now);
```

Comment:
This class shouldn't have a dependency on TimeProvider, the command handler should use it to get the time not the orchestrator.

Reply:
I disagree. The flaw in the current system is that we don't pass the date to the controller from the client.
When rendering the page we need to record the date/time of the most recent alert, that way when we dismiss we can ensure we only dismiss what the user has seen and not what they have seen + any that were created in the meantime.
This should really be added as a tech-debt.

Reply:
Add it as tech debt then please to the Tech Debt confluence page.